examples/fastapi/README.md

```diff
@@ -0,0 +1,126 @@
+# HumanLayer FastAPI Example
+
+This example demonstrates how to use AsyncHumanLayer with FastAPI to create an API with human-in-the-loop approval flows.
+
+## Overview
+
+The example shows how to:
+
+- Use AsyncHumanLayer with FastAPI endpoints
+- Handle human approvals for API operations
+- Process both approved and denied responses
+- Structure async API endpoints with human-in-the-loop flows
+
+## Setup
+
+1. Create a virtual environment and activate it:
+
+   ```bash
+   python -m venv venv
+   source venv/bin/activate  # On Windows: venv\Scripts\activate
+   ```
+
+2. Install dependencies:
+
+   ```bash
+   pip install -r requirements.txt
+   ```
+
+3. Copy `.env.example` to `.env` and fill in your API keys:
+   ```bash
+   cp .env.example .env
+   # Add your keys to .env:
+   # HUMANLAYER_API_KEY=your_key_here
+   # OPENAI_API_KEY=your_key_here
+   ```
+
+## Code Structure
+
+The main application (`app.py`) shows how to:
+
+1. Initialize AsyncHumanLayer:
+
+   ```python
+   from humanlayer import AsyncHumanLayer
+   hl = AsyncHumanLayer.cloud(verbose=True)
+   ```
+
+2. Create functions requiring approval:
+
+   ```python
+   @hl.require_approval()
+   async def multiply(a: int, b: int) -> int:
+       return a * b
+   ```
+
+3. Handle approvals in endpoints:
+   ```python
+   @app.post("/math/multiply")
+   async def math_multiply(a: int, b: int) -> Dict[str, str | int]:
+       result = await multiply(a, b)
```

Comment:
Today*, for the decorator to correctly identify to the backend, they should be passed as kwargs. 

```suggestion
       result = await multiply(a=a, b=b)
```

*we could probably change this